src/chapters/errors-and-debugging/categories-of-errors.rst

```diff
@@ -0,0 +1,146 @@
+
+During the first few weeks of your programming career, you will probably spend a lot of time tracking down syntax errors. However, as you gain experience, you will make fewer errors and you will also be able to find your errors faster.
+
+.. admonition:: Try It!
+
+   Find the syntax errors in the program.
+
+   .. sourcecode:: js
+
+      let day = Wednesday;
+      console.log(day;
+
+   `Run this program at repl.it <https://repl.it/@launchcode/Syntax-Errors>`_
+
+
+.. admonition:: Question
+
+   What syntax errors did you find? What was the specific error message provided by JavaScript in each case?
+
+Runtime Errors
+--------------
+
+.. index::
+   single: error; runtime
+
+.. index:: exception
+
+The second category of error is a **runtime error**, so called because the error does not appear until you run the program. These errors are also called **exceptions** because they usually indicate that something exceptional (and bad) has happened.
+
+Runtime errors occur during the execution phase of a program. This means that we will only encounter a runtime error once the syntax of our program has been validated. We will only see runtime errors after we have fixed any syntax errors. 
+
+A common runtime error occurs when we try to use a variable that has not been created yet. This can happen if you misspell the name of a variable, as the following example shows.
+
+.. admonition:: Example
+
+   .. sourcecode:: js
+      :linenos:
+   
+      let firstName = "Jack";
+      console.log(firstname);
+
+   **Output**
+
+   ::
+
+      ReferenceError: firstname is not defined
+         at evalmachine.<anonymous>:2:13
+         at Script.runInContext (vm.js:107:20)
+         at Object.runInContext (vm.js:285:6)
+         at evaluate (/run_dir/repl.js:133:14)
+         at ReadStream.<anonymous> (/run_dir/repl.js:116:5)
+         at ReadStream.emit (events.js:189:13)
+         at addChunk (_stream_readable.js:284:12)
+         at readableAddChunk (_stream_readable.js:265:11)
+         at ReadStream.Readable.push (_stream_readable.js:220:10)
+         at lazyFs.read (internal/fs/streams.js:181:12)
+
+The syntax of our program is correct, but when the program executes an error occurs when it reaches line 2. We attempt to print the value of the variable ``firstname``, but such a variable does not exist.
+
+Logic Errors
+------------
+
+.. index::
+   single: error; logic
+
+The third type of error is the **logic error**. If there is a logic error in your program, it will run successfully in the sense that the computer will not generate any error messages. However, the program will not work as intended.
+
+With a logic error, the problem is that the program you wrote is not the program you wanted to write. For example, say you were writing a program to calculate your daily earnings based on your weekly salary. You might write a program like the following:
```

Comment:
Rephrasing: "The characteristic of logic errors is that the program you wrote is not the program you wanted. For example, say you want a program to calculate your daily earnings based on your weekly salary. You might try the following:"

Alternative to 'characteristic of' ---> 'idea behind'.